tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -882,15 +875,77 @@ void SpirvBuilder::createBarrier(spv::Scope memoryScope,
   insertPoint->addInstruction(barrier);
 }
 
-SpirvBitFieldInsert *SpirvBuilder::createBitFieldInsert(
-    QualType resultType, SpirvInstruction *base, SpirvInstruction *insert,
-    SpirvInstruction *offset, SpirvInstruction *count, SourceLocation loc) {
+SpirvInstruction *SpirvBuilder::createEmulatedBitFieldInsert(
+    QualType ResultType, uint32_t BaseTypeBitwidth, SpirvInstruction *Dst,
+    SpirvInstruction *Src, unsigned BitOffset, unsigned BitCount,
+    SourceLocation loc, SourceRange range) {
+
+  assert(BitCount <= 64 &&
+         "Bitfield insertion emulation can only insert at most 64 bits.");
+  auto *DstTy = dyn_cast<IntegerType>(Dst->getResultType());
+  uint64_t MaskValue = 0;
+  for (unsigned i = 0; i < BitCount; i++)
```

Comment:
To create a mask of `n` 1s, I think you can simply do `2^n - 1` = `(1 << n) - 1`